src/aof.c

```diff
@@ -265,7 +417,45 @@ void flushAppendOnlyFile(int force) {
      * or alike */
 
     latencyStartMonitor(latency);
-    nwritten = write(server.aof_fd,server.aof_buf,sdslen(server.aof_buf));
+#ifdef USE_NVML
+    if (server.aof_use_nvml == 1) {
+        size_t len = sdslen(server.aof_buf);
+        size_t newlen = server.aof_current_size + len;
+
+        /* Check if there's required space, if not try to rewrite first */
+        if (newlen > server.aof_nvml_log_size)
```

Comment:
Ok, in original design there was case to trigger a rewrite at low aof space to compact current log. Also w didn't lose data, flushAppendOnlyFile is used to flush append buffer. If there was write error we leave function in 534 loc without clearing the buffer.